File: cyber/demo_py/demo05_param_server.py
from cyber.python.cyber_py3 import cyber
from cyber.python.cyber_py3 import parameter

if __name__ == "__main__":
    cyber.init()
    print("param service ....")
    node = cyber.Node("car_param")

    param_service = parameter.ParameterServer(node)

    param1 = parameter.Parameter("car_type", "car_apollo")
    param2 = parameter.Parameter("height", 1.5)
    param3 = parameter.Parameter("cameras", 2)

    param_service.set_parameter(param1)
    param_service.set_parameter(param2)
    param_service.set_parameter(param3)

    p1 = param_service.get_parameter("car_type")
    p2 = param_service.get_parameter("height")
    p3 = param_service.get_parameter("cameras")

    print(f"type = { p1.as_string() }, length = { p2.as_double() }, cameras = { p3.as_int64() }")

    ps = param_service.get_paramslist()
    for p in ps:
        print(p.debug_string())

    node.spin()

    cyber.shutdown()

File: cyber/demo_cc/demo05_param_server.cc
#include "cyber/cyber.h"
#include "cyber/parameter/parameter_server.h"

using apollo::cyber::ParameterServer;
using apollo::cyber::Parameter;

int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);
    AINFO << "parameter server";

    std::shared_ptr<apollo::cyber::Node> param_service_node = apollo::cyber::CreateNode("param");
    auto param_service = std::make_shared<ParameterServer>(param_service_node);

    param_service->SetParameter(Parameter("type", "apollo_car"));
    param_service->SetParameter(Parameter("height", 1.65));
    param_service->SetParameter(Parameter("lasers", 4));

    Parameter temp;
    param_service->GetParameter("type", &temp);
    AINFO << temp.Name() << ":" << temp.AsString();
    param_service->GetParameter("height", &temp);
    AINFO << temp.Name() << ":" << temp.AsDouble();
    param_service->GetParameter("lasers", &temp);
    AINFO << temp.Name() << ":" << temp.AsInt64();

    std::vector<Parameter> params;
    param_service->ListParameters(&params);
    for(auto &&param : params){
        AINFO << param.Name() << " = " << param.TypeName();
    }

    apollo::cyber::WaitForShutdown();
    return 0;
}
